Userland/Utilities/top.cpp

```diff
@@ -114,7 +134,49 @@ static Snapshot get_snapshot()
 static bool g_window_size_changed = true;
 static struct winsize g_window_size;
 
-int main(int, char**)
+bool parse_args(int argc, char** argv, TopOption& top_option)
+{
+    Core::ArgsParser::Option sort_by_option {
+        true,
+        "Sort by field [pid, tid, pri, user, state, virt, phys, cpu, name]",
+        "sort-by",
+        's',
+        nullptr,
+        [&top_option](const char* s) {
+            if (!strcmp(s, "pid"))
+                top_option.sort_by = TopOption::SortBy::pid;
+            else if (!strcmp(s, "tid"))
+                top_option.sort_by = TopOption::SortBy::tid;
+            else if (!strcmp(s, "pri"))
+                top_option.sort_by = TopOption::SortBy::priority;
+            else if (!strcmp(s, "user"))
+                top_option.sort_by = TopOption::SortBy::username;
+            else if (!strcmp(s, "state"))
+                top_option.sort_by = TopOption::SortBy::state;
+            else if (!strcmp(s, "virt"))
+                top_option.sort_by = TopOption::SortBy::virt;
+            else if (!strcmp(s, "phys"))
+                top_option.sort_by = TopOption::SortBy::phys;
+            else if (!strcmp(s, "cpu"))
+                top_option.sort_by = TopOption::SortBy::cpu;
+            else if (!strcmp(s, "name"))
+                top_option.sort_by = TopOption::SortBy::name;
+            else
+                return false;
+            return true;
+        }
+    };
+
+    Core::ArgsParser args_parser;
+
+    args_parser.set_general_help("Display information about processes");
+    args_parser.add_option(top_option.update_frequency, "Update frequency in seconds", "update-frequency", 'u', nullptr);
+    args_parser.add_option(move(sort_by_option));
+    args_parser.parse(argc, argv);
+    return true;
```

Comment:
Doesn't look like this can ever return `false`, and you're not using the return value; consider making this just `void`.

it's also pretty common in the project to just put the args parser directly in `main`, though I'm not particularly disagreeing with separating it into another function